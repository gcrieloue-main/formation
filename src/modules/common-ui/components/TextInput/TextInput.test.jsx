import React from 'react';
import TestRenderer from 'react-test-renderer';

import { TextInput } from '.';

it('renders correctly a TextInput component', () => {
  const testRenderer = TestRenderer.create(<TextInput />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it('renders correctly a TextInput component with onChange', () => {
  const mockOnChange = jest.fn();
  const testRenderer = TestRenderer.create(<TextInput placeHolder="placeholder" onChange={mockOnChange} />);
  const input = testRenderer.root.findByType('input');

  expect(input.props.placeholder).toBe('placeholder');

  input.props.onChange();

  expect(mockOnChange).toHaveBeenCalled();
});

it('renders correctly a TextInput component with password', () => {
  const mockOnChange = jest.fn();
  const testRenderer = TestRenderer.create(
    <TextInput placeHolder="placeholder" type="password" onChange={mockOnChange} />
  );
  const input = testRenderer.root.findByType('input');

  expect(input.props.type).toBe('password');
});
