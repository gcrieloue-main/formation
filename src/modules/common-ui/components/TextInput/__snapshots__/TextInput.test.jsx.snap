// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly a TextInput component 1`] = `
<span
  className="input"
>
  <label>
    
  </label>
  <input
    onKeyDown={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</span>
`;
